File: snippets/javascript/state-bag-handler.js
// CFX State Bag Change Handler (JavaScript)
// React to state changes on any entity or player bag.

AddStateBagChangeHandler('myResource:fuel', null, (bagName, key, value, _reserved, replicated) => {
    const entity = GetEntityFromStateBagName(bagName);
    if (entity === 0) return;

    if (replicated) {
        // change came from network; validate if needed
    }

    // react to the new value
});

File: snippets/lua/state-bag-handler.lua
-- CFX State Bag Change Handler
-- React to state changes on any entity or player bag.

AddStateBagChangeHandler('myResource:fuel', nil, function(bagName, key, value, _reserved, replicated)
    local entity = GetEntityFromStateBagName(bagName)
    if entity == 0 then return end

    if replicated then
        -- change came from network; validate if needed
    end

    -- react to the new value
end)
